src/phoenix/server/span_handler.py

```diff
@@ -35,5 +39,14 @@ async def post(self, request: Request) -> Response:
                 pb_span = encode(span)
         except Exception:
             return Response(status_code=422)
-        self.queue.put(pb_span)
+        version_number: Optional[float] = None
+        if version := request.headers.get("span_version"):
```

Comment:
Weird way to pass data - are you trying to not extend the protos? Did you consider alternatives? Ideally the data transport is fully encapsulated in the content and not on the header.

Reply:
Yea, that's because `version` is strictly optional, and the proto or json should have no knowledge of it. It's just part of how the data _can_ be processed and is not related to the data per se.